web/CMakeLists.txt

```diff
@@ -1,78 +1,45 @@
-  src/open-dialog.js
-  src/overflow-menu.js
-  src/prefs-dialog.js
-  src/prefs.js
-  src/remote.js
-  src/remove-dialog.js
-  src/rename-dialog.js
-  src/shortcuts-dialog.js
-  src/statistics-dialog.js
-  src/torrent.js
-  src/torrent-row.js
-  src/transmission.js
-  src/utils.js
-  style/transmission-app.scss
+file(GLOB_RECURSE TRWEB_ASSETS
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/img/*.png
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/img/*.svg
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/img/favicon.ico
 )
 
-set(TRWEB_IMGS
-  style/images/analytics.svg
-  style/images/application-x-executable.png
-  style/images/audio-x-generic.png
-  style/images/blue-turtle.png
-  style/images/checkered-flag.svg
-  style/images/chevron-down.svg
-  style/images/chevron-up.svg
-  style/images/diagram-3-fill.svg
-  style/images/files.svg
-  style/images/folder.png
-  style/images/font-x-generic.png
-  style/images/gear-fill.svg
-  style/images/horizontal-rule.svg
-  style/images/image-x-generic.png
-  style/images/lock-fill.svg
-  style/images/logo.png
-  style/images/package-x-generic.png
-  style/images/pause-circle-active.svg
-  style/images/pause-circle-idle.svg
-  style/images/play-circle-active.svg
-  style/images/play-circle-idle.svg
-  style/images/router.svg
-  style/images/search.svg
-  style/images/team.svg
-  style/images/text-x-generic.png
-  style/images/three-dots-vertical.svg
-  style/images/toolbar-close.png
-  style/images/toolbar-folder.png
-  style/images/toolbar-info.png
-  style/images/toolbar-pause.png
-  style/images/toolbar-start.png
-  style/images/turtle.png
-  style/images/up-and-down-arrows.svg
-  style/images/video-x-generic.png
+file(GLOB_RECURSE TRWEB_TMP_BUILD
+  ${CMAKE_CURRENT_SOURCE_DIR}/public_html/assets/**/*
+  ${CMAKE_CURRENT_SOURCE_DIR}/public_html/js/*
+  ${CMAKE_CURRENT_SOURCE_DIR}/public_html/*
 )
 
-add_custom_target(
-  trweb ALL
-  COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/package.json" "${CMAKE_CURRENT_BINARY_DIR}/package.json"
-  COMMAND ${CMAKE_COMMAND} -E copy "${CMAKE_CURRENT_SOURCE_DIR}/yarn.lock" "${CMAKE_CURRENT_BINARY_DIR}/yarn.lock"
-  COMMAND yarn install
-  COMMAND yarn webpack --config "${CMAKE_CURRENT_SOURCE_DIR}/webpack.config.js" --context "${CMAKE_CURRENT_SOURCE_DIR}"
-  BYPRODUCTS
-    public_html/transmission-app.js
-    public_html/transmission-app.js.LICENSE.txt
-  DEPENDS ${TRWEB_IMGS}
-  SOURCES ${TRWEB_SRCS}
+file(GLOB_RECURSE TRWEB_SRCS
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/*.js
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/index.html
+  ${CMAKE_CURRENT_SOURCE_DIR}/src/assets/css/*.scss
 )
+
+find_program(NODE node)
+
+if (NODE)
+    add_custom_target(
+      trweb ALL
+      COMMENT "Building Web"
+      COMMAND node --version
+      COMMAND npm config set package-lock false
+      COMMAND npm install --no-save corepack
+      COMMAND npx corepack enable
+      COMMAND npx yarn set version stable
+      COMMAND npx yarn install WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}"
```

Comment:
IIUC `WORKING_DIRECTORY` applies to all commands in this call. The way you have formatted the code right now seems to suggest that you expect only a single command to use it, which is misleading.

Reply:
@ckerr `WORKING_DIRECTORY` was your suggestion. Should we leave it?